src/Nrfutil/sandbox.ts

```diff
@@ -0,0 +1,420 @@
+            delete env.NRFUTIL_DEVICE_PLUGINS_DIR_FORCE_NRFDL_LOCATION;
+            delete env.NRFUTIL_DEVICE_PLUGINS_DIR_FORCE_NRFUTIL_LIBDIR;
+            delete env.NRFUTIL_IGNORE_MISSING_SUBCOMMAND;
+            delete env.NRFUTIL_LOG;
+            delete env.NRFUTIL_PACKAGE_INDEX_URL;
+        }
+
+        return env;
+    };
+
+    const processLoggingData = (data: NrfutilJson) => {
+        if (data.type === 'log') {
+            onLoggingHandlers.forEach(onLogging => onLogging(data.data));
+            return true;
+        }
+
+        return false;
+    };
+
+    const commonParser = <Result>(
+        data: Buffer,
+        callbacks: {
+            onProgress?: (progress: Progress) => void;
+            onInfo?: (info: Result) => void;
+            onTaskEnd?: (taskEnd: TaskEnd<Result>) => void;
+            onLogging?: (logging: LogMessage) => void;
+        }
+    ): Buffer | undefined => {
+        const parsedData: NrfutilJson<Result>[] | undefined =
+            parseJsonBuffers(data);
+
+        if (!parsedData) {
+            return data;
+        }
+
+        parsedData.forEach(item => {
+            switch (item.type) {
+                case 'task_progress':
+                    callbacks.onProgress?.(item.data.progress);
+                    break;
+                case 'task_end':
+                    callbacks.onTaskEnd?.(item.data);
+                    break;
+                case 'info':
+                    callbacks.onInfo?.(item.data);
+                    break;
+                case 'log':
+                    callbacks.onLogging?.(item.data);
+                    break;
+            }
+        });
+    };
+
+    const env = prepareEnv();
+
+    const getModuleVersion = (): Promise<ModuleVersion> =>
+        new Promise((resolve, reject) => {
+            execNrfutil<ModuleVersion>(module, ['--version'])
+                .then(results => {
+                    if (results.info.length === 1) {
+                        resolve(results.info[0]);
+                    } else {
+                        reject(new Error('Unexpected result'));
+                    }
+                })
+                .catch(reject);
+        });
+
+    const isSandboxInstalled = (): Promise<boolean> =>
+        new Promise((resolve, reject) => {
+            if (
+                fs.existsSync(
+                    path.join(
+                        baseDir,
+                        'nrfutil-sandboxes',
+                        module,
+                        version,
+                        'bin',
+                        `nrfutil-${module}${
+                            os.platform() === 'win32' ? '.exe' : ''
+                        }`
+                    )
+                )
+            ) {
+                getModuleVersion()
+                    .then(moduleVersion =>
+                        resolve(moduleVersion.version === version)
+                    )
+                    .then(reject);
```

Comment:
This probably was supposed to be `.catch(reject)`, right? The way it is currently, `resolve` and `reject` are always both called when the Promise returned by `getModuleVersion` resolves.